tests/network.md

```diff
@@ -546,3 +546,19 @@ EPIPE:
   Eio.Net.getnameinfo env#net sockaddr;;
 - : string * string = ("localhost", "http")
 ```
+
+## with_tcp_connet
+
+<!-- $MDX non-deterministic=command -->
```

Comment:
You can pass in a mock network and make this test actually run. We already tested that DNS lookups and connecting work, so we only need to test the logic of trying them in sequence here.

Reply:
It seems we can mock this just yet, since we also need to be able to mock `clock` due to - I think - usage of `Eio.Time.timeout` function.

Reply:
I've added a mock clock now in #328.